File: testfiles/prime.c
#include <stdio.h>

int findFirstPrime(int numbers[], int size) {
    for (int i = 0; i < size; i++) {
        int n = numbers[i];
        if (n < 2) {
            continue;
        }
        int isPrime = 1;
        int d = 2;
        while (d * d <= n) {
            if (n % d == 0) {
                isPrime = 0;
                break;
            }
            d = d + 1;
        }
        if (isPrime) {
            return n;
        }
    }
    return -1;
}

int classify(int score) {
    switch (score / 10) {
        case 10:
        case 9:
            return 'A';
        case 8:
            return 'B';
        default:
            return 'F';
    }
}

File: testfiles/prime.go
package main

func findFirstPrime(numbers []int) int {
	for i := 0; i < len(numbers); i++ {
		n := numbers[i]
		if n < 2 {
			continue
		}
		isPrime := true
		d := 2
		for d*d <= n {
			if n%d == 0 {
				isPrime = false
				break
			}
			d = d + 1
		}
		if isPrime {
			return n
		}
	}
	return -1
}

func classify(score int) string {
	switch {
	case score >= 90:
		return "A"
	case score >= 80:
		return "B"
	default:
		return "F"
	}
}
